base/linalg/symmetric.jl

```diff
@@ -0,0 +1,64 @@
+# Symmetric matrices
+
+type Symmetric{T<:Number} <: AbstractMatrix{T}
+    S::Matrix{T}
+    uplo::Char
+end
+function Symmetric{T<:Number}(S::Matrix{T}, uplo::Symbol)
+    if size(S, 1) != size(S, 2) throw(DimensionMismatch("Matrix must be square")); end
+    return Symmetric(S, string(uplo)[1])
+end
+Symmetric(A::StridedMatrix) = Symmetric(A, :U)
+
+copy(A::Symmetric) = Symmetric(copy(A.S), A.uplo)
+size(A::Symmetric, args...) = size(A.S, args...)
+print_matrix(io::IO, A::Symmetric) = print_matrix(io, full(A))
+full(A::Symmetric) = A.S
+ishermitian{T<:Real}(A::Symmetric) = true
+ishermitian{T<:Complex}(A::Symmetric) = all(imag(A.S) .== 0)
+issym(A::Symmetric) = true
+transpose(A::Symmetric) = A
+
+*(A::Symmetric, B::Symmetric) = *(full(A), full(B))
+*(A::Symmetric, B::StridedMatrix) = *(full(A), B)
```

Comment:
Was there a reason for the fallbacks on this and the next line, allocating a new array rather than calling `BLAS.symm!`?
Only the fallback from the line below are [still on master](https://github.com/JuliaLang/julia/blob/ff3a9aae6f847c9355e053d9b053140ce023a651/base/linalg/symmetric.jl#L297), should that fallback be removed? How would you else be able to dispatch [here](https://github.com/JuliaLang/julia/blob/ff3a9aae6f847c9355e053d9b053140ce023a651/base/linalg/symmetric.jl#L289-L290) or [here](https://github.com/JuliaLang/julia/blob/ff3a9aae6f847c9355e053d9b053140ce023a651/base/linalg/symmetric.jl#L293-L294)
cc @andreasnoack